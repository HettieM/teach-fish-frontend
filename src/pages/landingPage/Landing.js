import React from "react";
import Links from "../../components/global/links/Links";
import Heading from "../../components/global/heading/Heading";

const Landing = () => {
  return (
    <section>
      <Heading>Landing Page</Heading>
      <Links href="/about">Learn More</Links>
    </section>
  );
};

export default Landing;
