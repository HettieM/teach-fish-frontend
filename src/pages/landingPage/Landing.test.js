import React from "react";
import { render, screen } from "@testing-library/react";
import renderer from "react-test-renderer";
import "jest-styled-components";

import Landing from "./Landing";

test("Landing page component is styled", () => {
  const tree = renderer.create(<Landing />).toJSON();
  expect(tree).toMatchSnapshot();
  expect(tree).toHaveStyleRule("display", "flex");
  expect(tree).toHaveStyleRule("height", "80vh");
  expect(tree).toHaveStyleRule("justify-content", "space-between");
});

test("Join the movement button renders correctly", () => {
  render(<Landing />);
  screen.getByRole("button");
  screen.getByText("Join the Movement!");
});

test("Learn More About the App link renders correctly", () => {
  render(<Landing />);

  screen.getByRole("link");
  screen.getByText("Learn More About the App");
});
