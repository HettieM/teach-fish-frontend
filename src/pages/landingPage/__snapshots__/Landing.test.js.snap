// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Landing page component is styled 1`] = `
.c1 {
  font-family: "Josefin Sans",sans-serif;
  font-size: 2rem;
  color: navy;
  text-align: center;
}

.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  height: 80vh;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  margin: 0;
}

.c2 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  text-align: center;
}

.c3 {
  text-align: center;
}

@media (min-width:600px) {
  .c1 {
    font-size: 4rem;
  }
}

<section
  className="c0"
>
  <h1
    className="c1"
  >
    Enterprise Adventure
  </h1>
  <section
    className="c2"
  >
    <a
      className="makeStyles-links-1"
      href="/about"
      onClick={[Function]}
    >
      Learn More About the App
    </a>
    <button
      className="MuiButtonBase-root MuiButton-root MuiButton-contained makeStyles-root-2"
      disabled={false}
      onBlur={[Function]}
      onClick={[Function]}
      onDragLeave={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      onKeyUp={[Function]}
      onMouseDown={[Function]}
      onMouseLeave={[Function]}
      onMouseUp={[Function]}
      onTouchEnd={[Function]}
      onTouchMove={[Function]}
      onTouchStart={[Function]}
      tabIndex={0}
      type="button"
    >
      <span
        className="MuiButton-label"
      >
        Join the Movement!
      </span>
    </button>
  </section>
  <div
    className="c3"
  >
    <img
      alt="Teach a Man to Fish logo"
      src="logo.png"
    />
  </div>
</section>
`;
